presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -365,19 +368,39 @@ else if (skipFirstBatch && isFirst) {
                 isFirst = false;
             }
             else {
-                recordReader.readVector(0, vector);
-
-                ObjectVector objectVector = vector.toObjectVector(batchSize);
-                for (int i = 0; i < batchSize; i++) {
-                    assertTrue(iterator.hasNext());
-                    Object expected = iterator.next();
-
-                    Object actual = objectVector.vector[i];
-                    if (actual instanceof Slice) {
-                        actual = decodeSlice(type, (Slice) actual);
+                recordReader.readVector(type, 0, resultObject);
+
+                if (objectInspector instanceof PrimitiveObjectInspector) {
+                    Vector vector = (Vector) resultObject;
+                    ObjectVector objectVector = vector.toObjectVector(batchSize);
+                    for (int i = 0; i < batchSize; i++) {
+                        assertTrue(iterator.hasNext());
+                        Object expected = iterator.next();
+
+                        Object actual = objectVector.vector[i];
+                        if (actual instanceof Slice) {
+                            actual = decodeObject(type, (Slice) actual);
+                        }
+
+                        if (!Objects.equals(actual, expected)) {
```

Comment:
You'll see that I didn't touch this line if you tell github to ignore whitespace changes. I'll remove it.